cdap-app-fabric/src/main/java/io/cdap/cdap/app/job/SupportBundleJob.java

```diff
@@ -0,0 +1,471 @@
+    this.statusJson = statusJson;
+    retryServiceMap = new HashMap<>();
+    queue = new LinkedList<>();
+    runMetricsMap = new HashMap<>();
+    tasks = new JsonArray();
+    statusJson.add("tasks", tasks);
+    for (String serviceId : serviceList) {
+      queue.offer(serviceId);
+    }
+  }
+
+  /** Generates system log */
+  public void systemLogProcessing(String basePath, String systemLogPath) {
+    /** Adds system logs into file */
+    List<Future> futureList = new ArrayList<>();
+    Map<String, JsonObject> systemLogTaskMap = new HashMap<>();
+
+    String componentId = "services";
+    while (!queue.isEmpty()) {
+      String serviceId = queue.poll();
+      JsonObject task = initializeTask("systemlog-" + serviceId, "SystemLogSupportBundleTask");
+      systemLogTaskMap.put(serviceId, task);
+      Future<String> futureService =
+          executor.submit(
+              () -> {
+                updateTask(task, basePath, "IN_PROGRESS");
+                if (serviceId.equals("pipeline")) {
+                  try {
+                    supportBundleSystemLogTask.generateLogFile(
+                        cConf,
+                        logReader,
+                        LoggingContextHelper.getLoggingContext(
+                            "system",
+                            serviceId,
+                            "studio",
+                            ProgramType.valueOfCategoryName(componentId)),
+                        systemLogPath,
+                        serviceId + "-system-log.txt");
+                    retryServiceMap.remove(serviceId);
+                  } catch (Exception e) {
+                    LOG.warn(
+                        String.format("Retried three times for this serviceId %s ", serviceId), e);
+                    queueTaskAfterFailed(serviceId, task, basePath);
+                  }
+                } else {
+                  try {
+                    supportBundleSystemLogTask.generateLogFile(
+                        cConf,
+                        logReader,
+                        LoggingContextHelper.getLoggingContext(
+                            Id.Namespace.SYSTEM.getId(), componentId, serviceId),
+                        systemLogPath,
+                        serviceId + "-system-log.txt");
+                    retryServiceMap.remove(serviceId);
+                  } catch (Exception e) {
+                    LOG.warn(
+                        String.format("Retried three times for this serviceId %s ", serviceId), e);
+                    queueTaskAfterFailed(serviceId, task, basePath);
+                  }
+                }
+                return serviceId;
+              });
+      if (!retryServiceMap.containsKey(serviceId)) {
+        futureList.add(futureService);
+      }
+    }
+
+    completeProcessing(futureList, systemLogTaskMap, basePath, "systemlog");
+  }
+
+  /** Generates pipeline info file */
+  public void piplineInfoProcessing(
```

Comment:
Again, any task-specific code must be in task, job should do coordination and provide utility functions